server/backend/db_backend.go

```diff
@@ -554,27 +580,52 @@ func (self *MongoBackend) getTransactionList(
 	filter *models.TxsFilter,
 ) ([]*models.Transaction, error) {
 	var transactions []*models.Transaction
-	findQuery := bson.M{
-		"$or": []bson.M{
-			{"from": address},
-			{"to": address},
-		},
-		"created_at": bson.M{
-			"$gte": filter.FromTime,
-			"$lte": filter.ToTime,
-		},
-	}
-	if filter.InputDataEmpty != nil {
-		findQuery["input_data_empty"] = *filter.InputDataEmpty
-	}
-	err := self.mongo.C("Transactions").
-		Find(findQuery).
-		Sort("-created_at").
-		Skip(filter.Skip).
-		Limit(filter.Limit).
-		All(&transactions)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get tx list: %v", err)
+	if self.useTransactionsByAddress() {
+		var transactionsList []*models.TransactionsByAddress
+		findQuery := bson.M{
+			"address": address,
+			"created_at": bson.M{
+				"$gte": filter.FromTime,
+				"$lte": filter.ToTime,
+			},
+		}
+		err := self.mongo.C("TransactionsByAddress").
+			Find(findQuery).
+			Sort("-created_at").
+			Skip(filter.Skip).
+			Limit(filter.Limit).
+			All(&transactionsList)
```

Comment:
Can't we join this query against the `Transactions` collection to do it all at once? Or at least fetch them all in a batch to do it all in 2 calls?

Reply:
That could be tricky in mongo, but I'll run some tests with some mongo js (you could iterate through results inside mongo query and in theory it could be faster than golang->mongo->golang->mongo if we won't hit any suboptimal problems)

Reply:
It looks like mongo supports types joins natively. Is this feature available for us? https://docs.mongodb.com/manual/reference/operator/aggregation/lookup/

Reply:
Yeah looking into it (should be there), hope it would be faster